### PR TITLE
[WIP] delay-limit processor 

### PR DESCRIPTION
**Pull request by Libisch**
#73
In order to limit the time a process runs, Add `delay_limit.py` common processor.

### CODE REVIEW

File: datapackage_pipelines_mojp/common/processors/delay_limit.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    minutes = (current - self.start_time).total_seconds() / 60
+    return minutes
+  
+  def _filter_resource(self, resource_descriptor, resource_data):
+    delay_limit = self._parameters.get("stop_after_minutes")
+    for row in resource_data:
+      time_gap = self.get_time_gap()
+      row = self._filter_row(row)
+      if time_gap > delay_limit:
+        break
```
Comment (OriHoch) [inline datapackage_pipelines_mojp/common/processors/delay_limit.py]: you need to yield the row here
`yield row`

**Comment by Libisch** [comment]
@OriHoch 
Running `$ dpp run ./clearmash/download-related-entities`
Getting this:
```

INFO    :Main                            :Skipping redis connection, host:None, port:6379
INFO    :Main                            :RUNNING ./clearmash/download-related-entities
INFO    :Main                            :- ..datapackage_pipelines_mojp.common.processors.load_sql_resource
INFO    :Main                            :- ..datapackage_pipelines_mojp.common.processors.delay_limit
INFO    :Main                            :- ..datapackage_pipelines_mojp.clearmash.processors.download_related
INFO    :Main                            :- ..datapackage_pipelines_mojp.common.processors.delete
INFO    :Main                            :- dump.to_sql
INFO    :Main                            :- (sink)
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource: Traceback (most recent call last):
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/load_sql_resource.py", line 64, in <module>
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     LoadSqlResource.main()
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/datapackage_pipelines_mojp/common/processors/base_processors.py", line 54, in main
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     spew(*cls(*ingest()).spew())
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/datapackage_pipelines_mojp/common/processors/base_processors.py", line 57, in spew
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     self._datapackage, self._resources = self._process(self._datapackage, self._resources)
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/load_sql_resource.py", line 13, in _process
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     return super(LoadSqlResource, self)._process(*args, **kwargs)
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/datapackage_pipelines_mojp/common/processors/base_processors.py", line 67, in _process
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     "schema": self._get_schema()})
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:   File "/Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/load_sql_resource.py", line 21, in _get_schema
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource:     with open(self._parameters["datapackage"]) as f:
INFO    :Main                            :..datapackage_pipelines_mojp.common.processors.load_sql_resource: FileNotFoundError: [Errno 2] No such file or directory: '../data/clearmash/download-new-entities/datapackage.json'
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/load_sql_resource.py: 1
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/ENV/lib/python3.6/site-packages/datapackage_pipelines/manager/../lib/internal/sink.py: -9
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/ENV/lib/python3.6/site-packages/datapackage_pipelines/specs/../lib/dump/to_sql.py: -9
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/delay_limit.py: -9
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/clearmash/processors/download_related.py: -9
ERROR   :Main                            :FAILED /Users/libi/mojp-dbs-pipelines/clearmash/../datapackage_pipelines_mojp/common/processors/delete.py: -9
INFO    :Main                            :RESULTS:
INFO    :Main                            :FAILURE: ./clearmash/download-related-entities 
```

File: datapackage_pipelines_mojp/common/processors/delay_limit.py (hunk 1 above)
Comment (Libisch) [inline datapackage_pipelines_mojp/common/processors/delay_limit.py]: @OriHoch still no change... but maybe it's a local issue.

**Comment by OriHoch** [comment]
fixed in this commit: https://github.com/Beit-Hatfutsot/mojp-dbs-pipelines/commit/69783d863f00397f27d8b5383c404d0b2f87ebac (merged to master)

**Comment by OriHoch** [comment]
done in #85

**Pull request closed without merging**